### ISSUE TITLE
enable more tests in the vim/lua package set

### ISSUE THREAD

**Issue by teto** [NixOS/nixpkgs#267072]
## Issue description

At some point I had tried to run tests for a subset of the lua package set. It revealed dependency cycle between lua and busted.
I open this ticket to track the different issues and our progress on enabling tests in the lua package set.

The motivation came from different neovim plugins that broke on my (complex hackish neovim) setup. Were we to run their tests, I could troubleshoot these issues faster. 

One way for lua packages to help with the effort is to use "luarocks test" to run tests ( https://github.com/luarocks/luarocks/wiki/test) so that we can use a standardized command to run their tests:
- [x] https://github.com/lewis6991/gitsigns.nvim/issues/913

# planned tests
- [x] haskells-tools.nvim 
- [x] fzf-lua https://github.com/NixOS/nixpkgs/pull/382459/files
- [ ] nui.nvim https://github.com/MunifTanjim/nui.nvim IMPORTANT
- [ ] gitsigns.nvim
- [x] grug-far https://github.com/MagicDuck/grug-far.nvim (doing it at https://github.com/NixOS/nixpkgs/pull/387099)
- [ ] neorg
- [ ] quicker.nvim https://github.com/stevearc/quicker.nvim
- [x] lualine
- [ ] rocks-nvim https://github.com/NixOS/nixpkgs/pull/381036 pending a neovim PR
- [x] rocks-dev.nvim https://github.com/NixOS/nixpkgs/pull/417200
- [x] nvim-nio
- [ ] https://github.com/nvimtools/none-ls.nvim/
- [ ] https://github.com/Vigemus/iron.nvim
- [ ] https://github.com/julienvincent/hunk.nvim
- [ ] https://github.com/TimUntersberger/neogit
- [ ] neotest-playwright (first attempt at https://github.com/NixOS/nixpkgs/pull/319314)
- [ ] https://github.com/lukas-reineke/indent-blankline.nvim (convert to lua first)
- [x] neotest https://github.com/NixOS/nixpkgs/pull/315735
- [x] plenary.nvim https://github.com/NixOS/nixpkgs/pull/267065

TODO: I will update the ticket with the generated failure
basically if you set pkgs/development/lua-modules/overrides.nix
```
+  haskell-tools-nvim  = prev.haskell-tools-nvim.overrideAttrs(oa: {
+    doCheck = true;
+  });
```

**Comment by teto** [comment]
cc @khaneliman since you seem interested in the topic
